---
title: "Correction for Figure 6 and Figure S6"
author: "Nils Eling and Catalina Vallejos"
date: '`r Sys.Date()`'
output:
  html_document:
      code_folding: hide
      toc: true
      toc_float: 
        collapsed: false
      number_sections: false 
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Figure 6 

This script reproduces the analysis performed for Figure 6 and Figure S6.

First, we will load the genenames from biomart.

```{r}
library(biomaRt)

# Initialize mart
ensembl <- useMart("ensembl")

# Select dataset
ensembl <- useDataset(dataset = "mmusculus_gene_ensembl", 
                      mart = ensembl)

# Select gene ID and gene name
genenames <- getBM(attributes = c("ensembl_gene_id", "external_gene_name"),
                   mart = ensembl)
rownames(genenames) <- genenames$ensembl_gene_id
```

# Loading MCMC chains

Now, we will load the chains for CD4+ T cells from the different time-points.

```{r}
library(BASiCS)
MCMC_day2 <- BASiCS_LoadChain(RunName = "day2_cata_long", StoreDir = "../../MCMCs")
MCMC_day4 <- BASiCS_LoadChain(RunName = "day4_cata_long", StoreDir = "../../MCMCs") 
MCMC_day7 <- BASiCS_LoadChain(RunName = "day7_cata_long", StoreDir = "../../MCMCs")

#MCMC_day2 <- BASiCS_LoadChain(RunName = "day2_noPG_long", StoreDir = "../../MCMCs")
#MCMC_day4 <- BASiCS_LoadChain(RunName = "day4_noPG_long", StoreDir = "../../MCMCs") 
#MCMC_day7 <- BASiCS_LoadChain(RunName = "day7_noPG_long", StoreDir = "../../MCMCs")
```

# Convergence diagnostics

We use `BASiCS_diagPlot` in order to visualise the effective sample size (ESS)
obtained for gene-specific expression parameters. The ESS can be used to 
assess the mixing of the MCMC, with low ESS indicating poor mixing. 

```{r Convergence, fig.height=5, fig.width=15}
library(ggplot2)
library(cowplot)
p1 <- BASiCS_diagPlot(MCMC_day2)
p2 <- BASiCS_diagPlot(MCMC_day4)
p3 <- BASiCS_diagPlot(MCMC_day7)
plot_grid(p1, p2, p3, ncol = 3)
```


The figures above suggest good mixing for most genes, although low ESS is 
observed for lowly expressed genes. To avoid making inferences for those genes, 
subsequent analyses will only focus on those genes with ESS of at least 500. 

```{r}
inclgenes <- (p1$data[,2] + p2$data[,2] + p3$data[,2]) > 1500
```

After applying this inclusion criteria, `r sum(!inclgenes)` will be excluded 
(i.e. `r sum(inclgenes)` will be considered within the differential test). 

## Differential expression analysis

Differential expression and residual over-dispersion testing

```{r}
Test.day2_vs_day4 <- BASiCS_TestDE(Chain1 = MCMC_day4, Chain2 = MCMC_day2, 
                                   EpsilonM = 1, OffSet = TRUE, 
                                   GroupLabel1 = "Day4", GroupLabel2 = "Day2",
                                   Plot = FALSE, PlotOffset = FALSE,
                                   GenesSelect = inclgenes,
                                   EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)

Test.day2_vs_day4.logFC0 <- BASiCS_TestDE(Chain1 = MCMC_day4, Chain2 = MCMC_day2, 
                                          EpsilonM = 0, OffSet = TRUE,
                                          GroupLabel1 = "Day4", GroupLabel2 = "Day2",
                                          Plot = FALSE, PlotOffset = FALSE,
                                          GenesSelect = inclgenes,
                                          EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)

Test.day4_vs_day7 <- BASiCS_TestDE(Chain1 = MCMC_day7, Chain2 = MCMC_day4, 
                                   EpsilonM = 1, OffSet = TRUE, 
                                   GroupLabel1 = "Day7", GroupLabel2 = "Day4",
                                   Plot = FALSE, PlotOffset = FALSE,
                                   GenesSelect = inclgenes,
                                   EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)

Test.day4_vs_day7.logFC0 <- BASiCS_TestDE(Chain1 = MCMC_day7, Chain2 = MCMC_day4, 
                                          EpsilonM = 0, OffSet = TRUE,
                                          GroupLabel1 = "Day7", GroupLabel2 = "Day4",
                                          Plot = FALSE, PlotOffset = FALSE,
                                          GenesSelect = inclgenes,
                                          EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)

Test.day2_vs_day7 <- BASiCS_TestDE(Chain1 = MCMC_day7, Chain2 = MCMC_day2, 
                                   EpsilonM = 1, OffSet = TRUE, 
                                   GroupLabel1 = "Day7", GroupLabel2 = "Day2",
                                   Plot = FALSE, PlotOffset = FALSE,
                                   GenesSelect = inclgenes,
                                   EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)

Test.day2_vs_day7.logFC0 <- BASiCS_TestDE(Chain1 = MCMC_day7, Chain2 = MCMC_day2, 
                                   EpsilonM = 0, OffSet = TRUE, 
                                   GroupLabel1 = "Day7", GroupLabel2 = "Day2",
                                   Plot = FALSE, PlotOffset = FALSE,
                                   GenesSelect = inclgenes,
                                   EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)
```

Figure 6A.

```{r}
library(ggplot2)

# Find genes that are not DE in any condition
genes <- intersect(intersect(Test.day2_vs_day4.logFC0$TableDisp$GeneName[
                          Test.day2_vs_day4.logFC0$TableMean$ResultDiffMean == "NoDiff"], 
                   Test.day2_vs_day7.logFC0$TableDisp$GeneName[
                     Test.day2_vs_day7.logFC0$TableMean$ResultDiffMean == "NoDiff"]),
                   Test.day4_vs_day7.logFC0$TableDisp$GeneName[
                     Test.day4_vs_day7.logFC0$TableMean$ResultDiffMean == "NoDiff"
                   ])
length(genes)

df <- data.frame(delta = c(Test.day2_vs_day4.logFC0$TableDisp$Disp2[
                             match(genes, Test.day2_vs_day4.logFC0$TableDisp$GeneName)
                             ],
                           Test.day2_vs_day4.logFC0$TableDisp$Disp1[
                             match(genes, Test.day2_vs_day4.logFC0$TableDisp$GeneName)
                             ],
                           Test.day4_vs_day7.logFC0$TableDisp$Disp1[
                             match(genes, Test.day4_vs_day7.logFC0$TableDisp$GeneName)
                           ]),
                 Day = factor(c(rep("Day 2", length(genes)), rep("Day 4", length(genes)), 
                         rep("Day 7", length(genes))), levels = c("Day 2", "Day 4", "Day 7")))

#Plot boxplots
bp.delta <- ggplot(df) + geom_boxplot(aes(Day, log(delta), fill = Day), outlier.shape = NA) +
  scale_fill_manual(values = c("#fdd49e","#ef6548" ,"#7f0000")) + theme(legend.position = 'none')

ggsave("../../New_Figures_Results/Figure_6/Fig6_A_cata.pdf", height = 5, width = 7)
```

Figure 6B

```{r}
# Remove genes that are lowly expressed in all conditions from testing
genes <- !(is.na(MCMC_day2@parameters$epsilon[1,]) | 
             is.na(MCMC_day4@parameters$epsilon[1,]) |
            is.na(MCMC_day7@parameters$epsilon[1,]))
Test.day2_vs_day4.sets <- BASiCS_TestDE(Chain1 = MCMC_day4, Chain2 = MCMC_day2, 
                                   EpsilonM = 1, OffSet = TRUE,
                                   GroupLabel2 = "Day2", GroupLabel1 = "Day4",
                                   Plot = FALSE, PlotOffset = FALSE, 
                                   OrderVariable = "GeneName",
                                   GenesSelect = genes & inclgenes, 
                                   EFDR_M = 0.1,
                                   EFDR_D = 0.1, EFDR_R = 0.1)
Test.day4_vs_day7.sets <- BASiCS_TestDE(Chain1 = MCMC_day7, Chain2 = MCMC_day4, 
                                   EpsilonM = 1, OffSet = TRUE,
                                   GroupLabel2 = "Day4", GroupLabel1 = "Day7",
                                   Plot = FALSE, PlotOffset = FALSE, 
                                   OrderVariable = "GeneName",
                                   GenesSelect = genes & inclgenes, 
                                   EFDR_M = 0.1, EFDR_D = 0.1, EFDR_R = 0.1)
# 1st: up-up
genesinter <- intersect(
  Test.day2_vs_day4.sets$TableResDisp$GeneName[Test.day2_vs_day4.sets$TableResDisp$ResultDiffResDisp == "Day4+"],
  Test.day4_vs_day7.sets$TableResDisp$GeneName[Test.day4_vs_day7.sets$TableResDisp$ResultDiffResDisp == "Day7+"]
)
cur_df <- data.frame(epsilon = c(Test.day2_vs_day4.sets$TableResDisp$ResDisp2[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day2_vs_day4.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day4_vs_day7.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day4_vs_day7.sets$TableResDisp$GeneName )
                           ]),
                 Day = factor(c(rep("Day 2", length(genesinter)), rep("Day 4", length(genesinter)), 
                         rep("Day 7", length(genesinter))), levels = c("Day 2", "Day 4", "Day 7")),
                 Genes = rep(genesinter, 3)) 
plot.1 <- ggplot(cur_df, aes(group = Genes)) + 
  geom_line(aes(x = Day, y = epsilon, alpha = 0.3)) +
  theme(legend.position = 'none', axis.title.x = element_blank()) +
  scale_x_discrete(expand=c(0.05,0.05)) +
  annotate("text", x=2.5, y=4, label= paste(length(genesinter), " genes", sep = ""), size = 5) 

# 2nd: down-down
genesinter <- intersect(
  Test.day2_vs_day4.sets$TableResDisp$GeneName[Test.day2_vs_day4.sets$TableResDisp$ResultDiffResDisp == "Day2+"],
  Test.day4_vs_day7.sets$TableResDisp$GeneName[Test.day4_vs_day7.sets$TableResDisp$ResultDiffResDisp == "Day4+"]
)
cur_df <- data.frame(epsilon = c(Test.day2_vs_day4.sets$TableResDisp$ResDisp2[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day2_vs_day4.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day4_vs_day7.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day4_vs_day7.sets$TableResDisp$GeneName )
                           ]),
                 Day = factor(c(rep("Day 2", length(genesinter)), rep("Day 4", length(genesinter)), 
                         rep("Day 7", length(genesinter))), levels = c("Day 2", "Day 4", "Day 7")),
                 Genes = rep(genesinter, 3)) 
plot.2 <- ggplot(cur_df, aes(group = Genes)) + 
  geom_line(aes(x = Day, y = epsilon, alpha = 0.3)) +
  theme(legend.position = 'none', axis.title.x = element_blank()) +
  scale_x_discrete(expand=c(0.05,0.05)) +
  annotate("text", x=2.5, y=4, label= paste(length(genesinter), " genes", sep = ""), size = 5) 

# 3rd: down-up
genesinter <- intersect(
  Test.day2_vs_day4.sets$TableResDisp$GeneName[Test.day2_vs_day4.sets$TableResDisp$ResultDiffResDisp == "Day2+"],
  Test.day4_vs_day7.sets$TableResDisp$GeneName[Test.day4_vs_day7.sets$TableResDisp$ResultDiffResDisp == "Day7+"]
)
cur_df <- data.frame(epsilon = c(Test.day2_vs_day4.sets$TableResDisp$ResDisp2[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day2_vs_day4.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day4_vs_day7.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day4_vs_day7.sets$TableResDisp$GeneName )
                           ]),
                 Day = factor(c(rep("Day 2", length(genesinter)), 
                                rep("Day 4", length(genesinter)), 
                         rep("Day 7", length(genesinter))), 
                         levels = c("Day 2", "Day 4", "Day 7")),
                 Genes = rep(genesinter, 3)) 
plot.3 <- ggplot(cur_df, aes(group = Genes)) + 
  geom_line(aes(x = Day, y = epsilon, alpha = 0.3)) +
  theme(legend.position = 'none', axis.title.x = element_blank()) +
  scale_x_discrete(expand=c(0.05,0.05)) +
  annotate("text", x=2.5, y=4, label= paste(length(genesinter), " genes", sep = ""), size = 5) 

# 4th: up-down
genesinter <- intersect(
  Test.day2_vs_day4.sets$TableResDisp$GeneName[Test.day2_vs_day4.sets$TableResDisp$ResultDiffResDisp == "Day4+"],
  Test.day4_vs_day7.sets$TableResDisp$GeneName[Test.day4_vs_day7.sets$TableResDisp$ResultDiffResDisp == "Day4+"]
)
cur_df <- data.frame(epsilon = c(Test.day2_vs_day4.sets$TableResDisp$ResDisp2[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day2_vs_day4.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day2_vs_day4.sets$TableResDisp$GeneName)
                           ],
                           Test.day4_vs_day7.sets$TableResDisp$ResDisp1[
                             match(genesinter,Test.day4_vs_day7.sets$TableResDisp$GeneName )
                           ]),
                 Day = factor(c(rep("Day 2", length(genesinter)), rep("Day 4", length(genesinter)), 
                         rep("Day 7", length(genesinter))), levels = c("Day 2", "Day 4", "Day 7")),
                 Genes = rep(genesinter, 3)) 
plot.4 <- ggplot(cur_df, aes(group = Genes)) + 
  geom_line(aes(x = Day, y = epsilon, alpha = 0.3)) +
  theme(legend.position = 'none', axis.title.x = element_blank()) +
  scale_x_discrete(expand=c(0.05,0.05)) +
  annotate("text", x=2.5, y=4, label= paste(length(genesinter), " genes", sep = ""), size = 5) 

library(cowplot)
dist.plot <- plot_grid(plot.1, plot.2, plot.3, plot.4, ncol = 2, nrow = 2, align = 'hv')
dist.plot

ggsave("../../New_Figures_Results/Figure_6/Fig6_B_cata.pdf", height = 8, width = 8)
```

Figure 6C.
Check if genes are still DV.

```{r}
# Cxcr5 - ENSMUSG00000047880
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000047880",]

# Tigit - ENSMUSG00000071552
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000071552",]

# Tyk2 - ENSMUSG00000032175
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000032175",]
```

Tyk2 is not differentially variable anymore.
The trend points in the wrong direction.

Figure 6D.
Check if genes are still DV.

```{r}
# Ikzf4 - ENSMUSG00000002578
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000002578",]

# Ly6c1 - ENSMUSG00000079018
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000079018",]

# Tbx21 - ENSMUSG00000001444
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000001444",]
```

Ikfz4 and Ly6c1 are still differential variable.
Tbx21 is not but the trend is the same.

Visualize their gene expression.

```{r}
# Offset correction
OffSet <- function(MCMC1, MCMC2){
  OS <- median(rowSums(MCMC1@parameters$mu)/rowSums(MCMC2@parameters$mu))
  OS
}

OS <- OffSet(MCMC1 = MCMC_day2, MCMC2 = MCMC_day4)
cur_MCMC2 <- MCMC_day2
cur_MCMC2@parameters$mu <- MCMC_day2@parameters$mu / OS
cur_MCMC2@parameters$phi <- MCMC_day2@parameters$phi * OS

# Read in Data
Data.Day2 <- readRDS("../../Data/Data_day2.rds")
Data.Day4 <- readRDS("../../Data/Data_day4.rds")
Data.Day7 <- readRDS("../../Data/Data_day7.rds")

Counts.day2 <- BASiCS_DenoisedCounts(Data.Day2, cur_MCMC2)
Counts.day4 <- BASiCS_DenoisedCounts(Data.Day4, MCMC_day4)
rownames(Counts.day2) <- genenames[rownames(Counts.day2),2]
rownames(Counts.day4) <- genenames[rownames(Counts.day4),2]


# First genes that increase in variability between day 2 and day 4 
# Cxcr5 - Day4
df.gene <- data.frame(value = c(log10(Counts.day2["Cxcr5",] + 1),
                                log10(Counts.day4["Cxcr5",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Cxcr5 <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Cxcr5 expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position=c(0,1), legend.justification=c(0,1))+ ylim(c(-0.1,5))
# Tigit - Day4
df.gene <- data.frame(value = c(log10(Counts.day2["Tigit",] + 1),
                                log10(Counts.day4["Tigit",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Tigit <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Tigit expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position='none', legend.justification=c(0,1))+ ylim(c(-0.1,5))
# Tnfsf14 - Day4
df.gene <- data.frame(value = c(log10(Counts.day2["Tnfsf14",] + 1),
                                log10(Counts.day4["Tnfsf14",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Tnfsf14 <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Tnfsf14 expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position='none', legend.justification=c(0,1))+ ylim(c(-0.1,5))

# Now genes that decrease in variability between day 2 and day 4
# Ikzf4 - Day 2
df.gene <- data.frame(value = c(log10(Counts.day2["Ikzf4",] + 1),
                                log10(Counts.day4["Ikzf4",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Ikzf4 <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Ikzf4 expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position='none', legend.justification=c(0,1))+ ylim(c(-0.1,5))

# Cd28 - Day 2
df.gene <- data.frame(value = c(log10(Counts.day2["Cd28",] + 1),
                                log10(Counts.day4["Cd28",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Cd28 <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Cd28 expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position='none', legend.justification=c(0,1))+ ylim(c(-0.1,5))

# Jak2 - Day 2
df.gene <- data.frame(value = c(log10(Counts.day2["Jak2",] + 1),
                                log10(Counts.day4["Jak2",] + 1)),
                      Day = factor(c(rep("Day 2", ncol(Counts.day2)), 
                                      rep("Day 4", ncol(Counts.day4))),
                                    levels= c("Day 2", "Day 4")))
jitter.Jak2 <- ggplot(df.gene, aes(Day, value)) + 
  geom_boxplot(aes(fill = Day), alpha = 0.3, outlier.shape = NA, width=0.25)+
  geom_jitter(aes(fill = Day), width=0.25, shape=21) + ylab("log10(Jak2 expr)") + xlab("") +
  scale_color_manual(values = c("#fdd49e", "#ef6548")) + scale_fill_manual(values = c("#fdd49e", "#ef6548")) +
  theme(legend.position='none', legend.justification=c(0,1)) + ylim(c(-0.1,5))

# Ratio of Tbx21 cells
length(which(Counts.day2["Tbx21",] > 0))/ncol(Counts.day2)
jitter.final <- plot_grid(jitter.Cxcr5, jitter.Tigit, jitter.Tnfsf14, 
                          jitter.Ikzf4, jitter.Cd28, jitter.Jak2,
                          align = "hv", nrow = 2, ncol = 3, labels = c("C", NA, NA, "D", NA, NA))

ggsave("../../New_Figures_Results/Figure_6/Fig6_CD_cata.pdf", width = 12, height = 7)
```


Check if Cxcr5 and Tbx21 are still differentially expressed.

```{r}
# Tbx21 - ENSMUSG00000001444
Test.day2_vs_day4$TableMean[Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000001444",]
# NoDiff
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000001444",]
# NoDiff

Test.day4_vs_day7$TableMean[Test.day4_vs_day7$TableMean$GeneName == "ENSMUSG00000001444",]
# NoDiff
Test.day4_vs_day7$TableResDisp[Test.day4_vs_day7$TableResDisp$GeneName == "ENSMUSG00000001444",]
# Day 4 - NoDiff

# Cxcr5 - ENSMUSG00000047880
Test.day2_vs_day4$TableMean[Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000047880",]
# Day 4
Test.day2_vs_day4$TableResDisp[Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000047880",]
# Day 4

Test.day4_vs_day7$TableMean[Test.day4_vs_day7$TableMean$GeneName == "ENSMUSG00000047880",]
# NoDiff
Test.day4_vs_day7$TableResDisp[Test.day4_vs_day7$TableResDisp$GeneName == "ENSMUSG00000047880",]
# NoDiff

```

Create plot as in Figure 6E

```{r}
cur_df <- data.frame(epsilon = c(Test.day2_vs_day4$TableResDisp$ResDisp2[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day4$TableResDisp$ResDisp1[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day7$TableResDisp$ResDisp1[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day4$TableResDisp$ResDisp2[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000047880"],
  Test.day2_vs_day4$TableResDisp$ResDisp1[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000047880"],
  Test.day2_vs_day7$TableResDisp$ResDisp1[
  Test.day2_vs_day4$TableResDisp$GeneName == "ENSMUSG00000047880"]),
                     mu = c(Test.day2_vs_day4$TableMean$Mean2[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day4$TableMean$Mean1[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day7$TableMean$Mean1[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000001444"],
  Test.day2_vs_day4$TableMean$Mean2[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000047880"],
  Test.day2_vs_day4$TableMean$Mean1[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000047880"],
  Test.day2_vs_day7$TableMean$Mean1[
  Test.day2_vs_day4$TableMean$GeneName == "ENSMUSG00000047880"]),
  gene = c("Tbx21", "Tbx21", "Tbx21", "Cxcr5", "Cxcr5", "Cxcr5"),
  time_points = c("Day_2", "Day_4", "Day_7", "Day_2", "Day_4", "Day_7"))

df.arrows <- data.frame(x = cur_df[c(1,2,4,5),2],
                        y = cur_df[c(1,2,4,5),1],
                        xend = cur_df[c(2,3,5,6),2],
                        yend = cur_df[c(2,3,5,6),1],
                        Genes = c("Tbx21", "Tbx21","Cxcr5", "Cxcr5"))

trace.plot <- ggplot(cur_df) + 
  geom_segment(data=df.arrows, 
        aes(x=log(x), y=y, xend=log(xend), yend=yend, colour = Genes),
        lineend="round",arrow=arrow(length=unit(0.2, "inches")), size=2) +
  xlab("log(Expr)") + ylab("Epsilon \n Less variable <-> More variable") +
  scale_colour_manual(values = setNames(c("indianred4", "royalblue4"), 
                                        c("Cxcr5", "Tbx21")))

trace.plot

ggsave("../../New_Figures_Results/Figure_6/Fig6_E_cata.pdf", trace.plot, height = 8, width = 8)
```

```{r, eval = FALSE}
TableDay2_4 <- merge(Test.day2_vs_day4$TableMean,
                     Test.day2_vs_day4$TableResDisp,
                     by = "GeneName")
TableDay2_4$MeanOverall <- TableDay2_4$MeanOverall.x
TableDay2_4$MeanOverall.x <- NULL
TableDay2_4$MeanOverall.y <- NULL
TableDay2_4$name <- genenames[TableDay2_4$GeneName, 2]

TableDay4_7 <- merge(Test.day4_vs_day7$TableMean,
                     Test.day4_vs_day7$TableResDisp,
                     by = "GeneName")
TableDay4_7$MeanOverall <- TableDay4_7$MeanOverall.x
TableDay4_7$MeanOverall.x <- NULL
TableDay4_7$MeanOverall.y <- NULL
TableDay4_7$name <- genenames[TableDay4_7$GeneName, 2]

write.csv(TableDay2_4, 
          "../../New_Figures_Results/TableDay2_4.csv")
write.csv(TableDay4_7, 
          "../../New_Figures_Results/TableDay4_7.csv")
```

# Figure S6

This script aims to dissect the differentiation process of CD4 T cells in more detail.
We will use genes annotated in Lönnberg et al.

## Figure S6A

We will use the gene modules detected by GPfates as displayed in Figure 3E in
Lönnberg et al.

```{r}
Th1 <- c("Cxcr6", "Gna15", "S100a4", "Gzmb", "Chsy1", "Ccr2", "Ccr5", "Ifngr1", "Sdcbp2",
         "Gp49a", "Plek", "Lgals3", "Havcr2", "Sytl2", "F2rl2", "Nkg7", "Il18rap", "Ctsw",
         "Entpd1", "Cd38", "Myo1f", "F2r", "Xdh", "Lilrb4", "Ahnak", "S100a6", "Serpinb6b",
         "Pglyrp1", "Ccl5", "Tmprss13", "Il12rb2", "Sla2", "Il18r1", "Ifng", "Ctsd",
         "S1pr1", "Tmem154", "Capn2", "Syt11", "Mfsd10", "Cd44", "Sema4a", "Ccl4",
         "Emp3", "4930453N24Rik", "Plac8", "Prex1", "Anxa2", "AU020206", "Glrx")
Tfh <- c("Asap1", "Tcf7", "Rgs10", "Tbc1d4", "Gm10275", "Zfp36l1", "Cxcr5",
         "Ddx24", "Rps12-ps3", "Rps20", "Erdr1", "Tnfsf8", "Cdk2ap2", "Kcnn4",
         "Rpl10a", "D19Bwg1357e", "Ivns1abp", "Bcas2", "Znf512b", "Tagap1",
         "Ptma", "Nop10", "Tnfrsf26", "Alyref", "Rpl36a", "Npm1", "Dennd2d", "Lsm7",
         "Il16", "Trp53", "Clec2i", "Rpl10", "Sarnp",
         "Srsf2", "Inpp4b", "Rps2", "Emg1", "Phb", "Hnrnpr", "Nap1l1", "Cops6",
         "Prdx6", "Lsm3", "Adk", "Mrps6", "Cdk4")

genes.ENS.Th1 <- genenames[match(Th1, genenames[,2]),1]
genes.ENS.Tfh <- genenames[match(Tfh, genenames[,2]),1]
genes.ENS.Th1 <- genes.ENS.Th1[!is.na(match(genes.ENS.Th1, Test.day2_vs_day4$TableMean$GeneName))]
genes.ENS.Tfh <- genes.ENS.Tfh[!is.na(match(genes.ENS.Tfh, Test.day2_vs_day4$TableMean$GeneName))]

final.genes <- c(genes.ENS.Th1, rev(genes.ENS.Tfh))

DE.2_4 <- ifelse(Test.day2_vs_day4$TableMean[match(final.genes,
                                Test.day2_vs_day4$TableMean$GeneName),"ResultDiffMean"] == "Day2+", "Down",
                 ifelse(Test.day2_vs_day4$TableMean[match(final.genes,
                                Test.day2_vs_day4$TableMean$GeneName),"ResultDiffMean"] == "Day4+", "Up",
                       "NoDiff"))

DE.4_7 <- ifelse(Test.day4_vs_day7$TableMean[match(final.genes,
                                Test.day4_vs_day7$TableMean$GeneName),"ResultDiffMean"] == "Day4+", "Down",
                 ifelse(Test.day4_vs_day7$TableMean[match(final.genes,
                                Test.day4_vs_day7$TableMean$GeneName),"ResultDiffMean"] == "Day7+", "Up",
                       "NoDiff"))

DV.2_4 <- ifelse(Test.day2_vs_day4$TableResDisp[match(final.genes,
                                Test.day2_vs_day4$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day2+", "Down",
                 ifelse(Test.day2_vs_day4$TableResDisp[match(final.genes,
                                Test.day2_vs_day4$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day4+", "Up",
                       "NoDiff"))

DV.4_7 <- ifelse(Test.day4_vs_day7$TableResDisp[match(final.genes,
                                Test.day4_vs_day7$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day4+", "Down",
                 ifelse(Test.day4_vs_day7$TableResDisp[match(final.genes,
                                Test.day4_vs_day7$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day7+", "Up",
                       "NoDiff"))

df <- data.frame(Day = rep(c("Day 2 vs Day 4", "Day 4 vs Day 7"), each = length(final.genes)),
                 Genes = factor(rep(genenames[final.genes,2], 2), levels = rev(genenames[final.genes,2])),
                 DE = c(DE.2_4, DE.4_7),
                 DV = c(DV.2_4, DV.4_7))

DE.module <- ggplot(df) + geom_point(aes(Day, Genes, colour = DE), size = 4) + coord_fixed(ratio = 1) + 
  scale_color_manual(values = c("blue", "grey", "red")) + theme(axis.text.x = element_text(angle = 90))

DV.module <- ggplot(df) + geom_point(aes(Day, Genes, colour = DV), size = 4) + coord_fixed(ratio = 1) +  
  scale_color_manual(values = c("green", "grey", "purple")) + theme(axis.text.x = element_text(angle = 90))

ggsave("../../New_Figures_Results/Figure_S6/FigS6_A.DE_cata.pdf", 
       DE.module, height = 12, width = 3)
ggsave("../../New_Figures_Results/Figure_S6/FigS6_A.DV_cata.pdf", 
       DV.module, height = 12, width = 3)
```

## Figure S6B

We will now take a closer look into the fate decision module

```{r}
# Important genes ordered by peak expression
stage <- c("Il2ra", "Tbx21", "Il2rb", "Cxcr5", "Cxcr3", "Il21", "Pdcd1", "Cxcr6", "Selplg",
            "Ccr2", "Id2", "Ifng", "Icos", "Ifngr1", "Ccr5")
stage.ENS <- genenames[match(stage, genenames[,2]),1]
stage.ENS <- stage.ENS[!is.na(match(stage.ENS, Test.day2_vs_day4$TableMean$GeneName))]
final.genes <- stage.ENS

DE.2_4 <- ifelse(Test.day2_vs_day4$TableMean[match(final.genes,
                                Test.day2_vs_day4$TableMean$GeneName),"ResultDiffMean"] == "Day2+", "Down",
                 ifelse(Test.day2_vs_day4$TableMean[match(final.genes,
                                Test.day2_vs_day4$TableMean$GeneName),"ResultDiffMean"] == "Day4+", "Up",
                       "NoDiff"))

DE.4_7 <- ifelse(Test.day4_vs_day7$TableMean[match(final.genes,
                                Test.day4_vs_day7$TableMean$GeneName),"ResultDiffMean"] == "Day4+", "Down",
                 ifelse(Test.day4_vs_day7$TableMean[match(final.genes,
                                Test.day4_vs_day7$TableMean$GeneName),"ResultDiffMean"] == "Day7+", "Up",
                       "NoDiff"))

DV.2_4 <- ifelse(Test.day2_vs_day4$TableResDisp[match(final.genes,
                                Test.day2_vs_day4$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day2+", "Down",
                 ifelse(Test.day2_vs_day4$TableResDisp[match(final.genes,
                                Test.day2_vs_day4$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day4+", "Up",
                       "NoDiff"))

DV.4_7 <- ifelse(Test.day4_vs_day7$TableResDisp[match(final.genes,
                                Test.day4_vs_day7$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day4+", "Down",
                 ifelse(Test.day4_vs_day7$TableResDisp[match(final.genes,
                                Test.day4_vs_day7$TableResDisp$GeneName),"ResultDiffResDisp"] == "Day7+", "Up",
                       "NoDiff"))
df <- data.frame(Day = rep(c("Day 2 vs Day 4", "Day 4 vs Day 7"), each = length(final.genes)),
                 Genes = factor(rep(genenames[final.genes,2], 2), levels = rev(genenames[final.genes,2])),
                 DE = c(DE.2_4, DE.4_7),
                 DV = c(DV.2_4, DV.4_7))

DE.selected <- ggplot(df) + geom_point(aes(Day, Genes, colour = DE), size = 4) + coord_fixed(ratio = 1) + 
  scale_color_manual(values = c("blue", "grey", "red")) + theme(axis.text.x = element_text(angle = 90))

DV.selected <- ggplot(df) + geom_point(aes(Day, Genes, colour = DV), size = 4) + coord_fixed(ratio = 1) +  
  scale_color_manual(values = c("green", "grey", "purple")) + theme(axis.text.x = element_text(angle = 90))

ggsave("../../New_Figures_Results/Figure_S6/FigS6_B.DE_cata.pdf", 
       DE.selected, height = 6, width = 3)
ggsave("../../New_Figures_Results/Figure_S6/FigS6_B.DV_cata.pdf", 
       DV.selected, height = 6, width = 3)
```


